repl/src/main/scala/com/cloudera/livy/repl/ReplDriver.scala

```diff
@@ -83,4 +95,40 @@ class ReplDriver(conf: SparkConf, livyConf: RSCConf)
     return session.state.toString
   }
 
+  override def createWrapper(msg: BaseProtocol.BypassJobRequest): BypassJobWrapper = {
+    kind match {
+      case Spark() => super.createWrapper(msg)
+      case PySpark() | PySpark3() => {
+        new BypassPySparkJobWrapper(this, msg.id,
+          new BypassPySparkJob(msg.serializedJob, pysparkJobProcessor))
+      }
+    }
+  }
+
+  def getEquivalentPySparkJobInstance(job: Job[_]): Job[_] = {
```

Comment:
Hmm, this is a little nasty... I think this would be better handled differently.

Since `AddFileJob` and `AddJarJob` are internal to the RSC, they can make use of `JobContextImpl`. If you add a reference to the `RSCDriver` instance there, and add new methods `addFile` and `addJar`, then you can dispatch them to the appropriate handler easily by overriding them here in `ReplDriver`.

Then you wouldn't even need the python-specific implementations of the jobs.